src/app/keycard/core.nim

```diff
@@ -0,0 +1,83 @@
+import NimQml, chronicles, std/wrapnils
+import status/[signals, status, keycard]
+import types/keycard as keycardtypes
+import view, pairing
+
+logScope:
+  topics = "keycard-model"
+
+type KeycardController* = ref object
+  view*: KeycardView
+  variant*: QVariant
+  status: Status
+
+proc newController*(status: Status): KeycardController =
+  result = KeycardController()
+  result.status = status
+  result.view = newKeycardView(status)
+  result.variant = newQVariant(result.view)
+
+proc delete*(self: KeycardController) =
+  delete self.variant
+  delete self.view
+
+proc reset*(self: KeycardController) =
+  discard
+
+proc attemptOpenSecureChannel(self: KeycardController) : bool =
+  let pairing = self.view.pairings.getPairing(self.view.appInfo.instanceUID)
+
+  if pairing == "":
+    return false
+
+  discard """let err = self.status.keycard.openSecureChannel(pairing)
+
+  if err == Ok:
+    return true
+
+  self.view.pairings.removePairing(self.view.appInfo.instanceUID)
+
+  """"
```

Comment:
If this is commented out entirely, we should remove this.

Reply:
thanks a lot for the review! I will address the issues tomorrow. The discarded blocks and empty slots are stubs (either empty or with some sort of pseudocode) that are going to be implemented. I added them to better visualize the flows but I can remove them before merging if you think it's better. Eventually these are all going to be implemented, ideally in separate PRs to keep it manageable.